File: src/main/java/pl/training/workshop/m6/s11_safecomposite/step2/MenuComponent.java
package pl.training.workshop.m6.s11_safecomposite.step2;

import pl.training.workshop.shared.Money;

/** Krok 2: forma Java 25 - zamknięty zestaw węzłów, rekordy, brak mutacji po zbudowaniu. */
public sealed interface MenuComponent permits Product, Combo {
    String name();

    Money price();

    String describe();
}

File: csharp/src/Training.Workshop/M6/S11SafeComposite/Step2/MenuComponent.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S11SafeComposite.Step2;

/// <summary>
/// Krok 2: forma C# odpowiadająca sealed interface z Javy - zamknięty zestaw węzłów
/// (konstruktor private protected: podtypy tylko w tym projekcie), rekordy, brak mutacji po zbudowaniu.
/// </summary>
public abstract record MenuComponent
{
    private protected MenuComponent()
    {
    }

    public abstract string Name { get; }

    public abstract Money Price { get; }

    public abstract string Describe();
}
